internal/dataplane/sendconfig/inmemory_schema.go

```diff
@@ -0,0 +1,124 @@
+package sendconfig
+
+import (
+	"github.com/kong/deck/file"
+)
+
+type ConsumerGroupConsumerRelationship struct {
+	ConsumerGroup string `json:"consumer_group"`
+	Consumer      string `json:"consumer"`
+}
+
+type ConsumerGroupPluginRelationship struct {
+	ConsumerGroup string `json:"consumer_group"`
+	Plugin        string `json:"plugin"`
+}
+
+type DBLessConfig struct {
+	file.Content                       `json:",inline"`
+	ConsumerGroupConsumerRelationships []ConsumerGroupConsumerRelationship `json:"consumer_group_consumers,omitempty"`
+	ConsumerGroupPluginRelationships   []ConsumerGroupPluginRelationship   `json:"consumer_group_plugins,omitempty"`
+}
+
+type DefaultContentToDBLessConfigConverter struct{}
+
+func (DefaultContentToDBLessConfigConverter) Convert(content *file.Content) DBLessConfig {
+	dblessConfig := DBLessConfig{
+		Content: *content,
+	}
+
+	// DBLess schema does not support decK's Info section.
+	dblessConfig.Content.Info = nil
+
+	// DBLess schema does not support nulls in plugin configs.
+	cleanUpNullsInPluginConfigs(&dblessConfig.Content)
+
+	// DBLess schema does not 1-1 match decK's schema for ConsumerGroups.
+	cleanupConsumerGroups(&dblessConfig)
```

Comment:
Maybe a different name than `cleanup` since it's more like a conversion IMHO. WDYT?